eZ/Publish/Core/Repository/Values/User/Policy.php

```diff
@@ -33,4 +33,19 @@ public function getLimitations()
     {
         return $this->limitations;
     }
+
+    /**
+     * Return true if the scope of current Policy object is more narrow than the scope of $policy.
+     *
+     * @param \eZ\Publish\Core\Repository\Values\User\Policy $policy
+     * @return bool
+     */
+    public function isNarrow(Policy $policy)
+    {
+        return
+            !empty($this->limitations) && empty($policy->getLimitations()) &&
+            ($this->module === $policy->module || $policy->module === '*') &&
+            ($this->function === $policy->function || $policy->function === '*')
+        ;
```

Comment:
I'm not a big fan, name is wrong ;)

Also I would prefer if we could handle the logic on SPI objects which are more light before `buildDomainPolicyObject` is called.